src/VisualStudio/Core/Impl/ProjectSystem/CPS/CPSProjectFactory.cs

```diff
@@ -94,5 +100,33 @@ private IVsReportExternalErrors GetExternalErrorReporter(ProjectId projectId, st
 
             return new ProjectExternalErrorReporter(projectId, errorCodePrefix, _serviceProvider);
         }
+
+        private static void EnsurePackageLoaded(IServiceProvider serviceProvider, string language)
+        {
```

Comment:
Can you change the CPSProjectFactory to be a Foreground affinitized object and ensure this is invoked on a Foreground thread?